Fix Pharmacy Union paying 3MC for its free card

Lynesth opened the pull request:
The free Science card is not its first action but is part of its `play()` method.
Therefore, that card was taken into account when calculating cost of initial hand of cards, effectively making the player start with 43MC instead of 46.

bafolts commented on src/Game.ts:
```diff
@@ -797,12 +797,12 @@ export class Game implements ILoadable<SerializedGame, Game> {
       player.megaCredits = corporationCard.startingMegaCredits;
       if (corporationCard.name !== new BeginnerCorporation().name) {
         const cardsToPayFor: number = player.cardsInHand.length;
         player.megaCredits -= cardsToPayFor * player.cardCost;
```
I don't think that this change will be this straight forward. The `TerralabsResearch` corporation has a `cardCost` of `1` that it needs set prior to charging for cards. Similarly `Polyphemos` has a `cardCost` of `5`. Any corporation which manipulates the card cost has to be applied prior to charging for cards.

bafolts replied:
If you use the local variable `cardCost` instead of `player.cardCost` this change should work.

Lynesth replied:
Fixed.

bafolts reviewed (changes requested):
I am not sure what the solution is yet but this will break corporations which modify `cardCost`.

Lynesth commented:
Fixed the issue that arised with **Terralabs Research** and **Polyphemos**.
Also added tests to all 3 (including **Pharmacy Union**) to check for starting hand card cost.
